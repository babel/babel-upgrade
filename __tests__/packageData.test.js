const { updatePackageJSON } = require('../src/');
const upgradeDeps = require('../src/upgradeDeps');
const babelCoreFixture = require('../fixtures/babel-core');
const jestFixture = require('../fixtures/jest');
const jestCliFixture = require('../fixtures/jest-cli');
const depsFixture = require('../fixtures/deps');
const webpackV1Fixture = require('../fixtures/webpack-v1');
const depsFixtureEarlierBeta = require('../fixtures/deps-earlier-beta.json');
const scriptsMochaFixture = require('../fixtures/scripts-mocha');
const scriptsBabelNodeFixture = require('../fixtures/scripts-babel-node');

const VERSION = "7.0.0-beta.39";

describe('upgradeDeps', () => {
  test('upgrades from v6', () => {
    expect(upgradeDeps(depsFixture, VERSION)).toMatchSnapshot();
  });

  test('upgrades from earlier v7 version', () => {
    expect(upgradeDeps(depsFixtureEarlierBeta, VERSION)).toMatchSnapshot();
  });

  test('splits runtime into runtime and runtime-corejs2', () => {
    expect(upgradeDeps({
      "@babel/runtime": "7.0.0-alpha.0",
    }, VERSION)).toMatchSnapshot();
  });

  test("doesn't add runtime-corejs2 if it is already there", () => {
    expect(upgradeDeps({
      "@babel/runtime": "7.0.0-alpha.0",
      "@babel/runtime-corejs2": "7.0.0-alpha.0",
    }, VERSION)).toMatchSnapshot();
  });
});

test('scripts', async () => {
  expect(await updatePackageJSON(scriptsMochaFixture)).toMatchSnapshot();
  expect(await updatePackageJSON(scriptsBabelNodeFixture)).toMatchSnapshot();
});

test('@babel/core peerDep', async () => {
  expect(await updatePackageJSON(babelCoreFixture)).toMatchSnapshot();
});

test('jest babel-core bridge', async () => {
  expect(await updatePackageJSON(jestFixture)).toMatchSnapshot();
});

test('jest-cli babel-core bridge', async () => {
  expect(await updatePackageJSON(jestCliFixture)).toMatchSnapshot();
});

test('webpack v1 compatibility', async () => {
  expect(await updatePackageJSON(webpackV1Fixture)).toMatchSnapshot();
});

test('replaces stage presets', () => {
  expect(upgradeDeps({
    "@babel/preset-stage-2": "7.0.0-alpha.0"
  }, VERSION)).toMatchSnapshot();
});
