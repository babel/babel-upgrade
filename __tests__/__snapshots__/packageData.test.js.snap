// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@babel/core peerDep 1`] = `
Object {
  "devDependencies": Object {
    "@babel/core": "7.0.0-beta.39",
    "@babel/plugin-proposal-object-rest-spread": "7.0.0-beta.39",
    "@babel/preset-env": "7.0.0-beta.39",
    "babel-loader": "v8.0.0-beta.0",
  },
  "name": "add-@babel/core-peerDep",
}
`;

exports[`jest babel-core bridge 1`] = `
Object {
  "devDependencies": Object {
    "@babel/core": "7.0.0-beta.39",
    "@babel/plugin-proposal-object-rest-spread": "7.0.0-beta.39",
    "@babel/preset-env": "7.0.0-beta.39",
    "babel-core": "^7.0.0-bridge.0",
    "babel-loader": "v8.0.0-beta.0",
    "jest": "22.0.0",
  },
  "name": "add-@babel/core-peerDep",
}
`;

exports[`scripts 1`] = `
Object {
  "name": "mocha-scripts-test",
  "scripts": Object {
    "coverage": "NODE_ENV=test nyc --reporter=lcov --reporter=text _mocha --require @babel/register test/*.js",
    "prepublish": "npm run test",
    "test": "mocha --compilers js:@babel/register --require @babel/polyfill test/*Test.js",
    "test2": "mocha --compilers js:@babel/register --require @babel/polyfill test/*Test.js",
  },
}
`;

exports[`scripts 2`] = `
Object {
  "devDependencies": Object {
    "@babel/cli": "7.0.0-beta.39",
    "@babel/node": "7.0.0-beta.39",
  },
  "name": "scripts-babel-node",
  "scripts": Object {
    "test": "babel-node",
  },
}
`;

exports[`upgradeDeps adds flow preset if user is upgrading from previous v7 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-flow": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps adds flow preset if user was using v6 preset-react 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-flow": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps does not add flow preset if user is upgrading from previous v7 but flow not detected 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps does not add flow preset if user was using v6 preset-react but flow not detected 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps handles flow preset if user had entry and is upgrading from previous v7 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-flow": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps handles flow preset if user had entry and is upgrading from previous v7 and flow not detected 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/preset-flow": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps upgrades from earlier v7 version 1`] = `
Object {
  "@babel/core": "7.0.0-beta.39",
  "@babel/plugin-proposal-object-rest-spread": "7.0.0-beta.39",
  "@babel/plugin-syntax-dynamic-import": "7.0.0-beta.39",
  "@babel/polyfill": "https://github.com/some/tarball/stable#85a1b40e4616fe54e4e13fdd9350128d1f0508e6",
  "@babel/preset-env": "7.0.0-beta.39",
  "@babel/preset-flow": "not/semver#dddddd",
  "@babel/preset-react": "7.0.0-beta.39",
}
`;

exports[`upgradeDeps upgrades from v6 1`] = `
Object {
  "@babel/cli": "7.0.0-beta.39",
  "@babel/code-frame": "7.0.0-beta.39",
  "@babel/core": "7.0.0-beta.39",
  "@babel/generator": "7.0.0-beta.39",
  "@babel/helper-annotate-as-pure": "7.0.0-beta.39",
  "@babel/helper-bindify-decorators": "7.0.0-beta.39",
  "@babel/helper-builder-binary-assignment-operator-visitor": "7.0.0-beta.39",
  "@babel/helper-builder-react-jsx": "7.0.0-beta.39",
  "@babel/helper-call-delegate": "7.0.0-beta.39",
  "@babel/helper-define-map": "7.0.0-beta.39",
  "@babel/helper-explode-assignable-expression": "7.0.0-beta.39",
  "@babel/helper-explode-class": "7.0.0-beta.39",
  "@babel/helper-fixtures": "7.0.0-beta.39",
  "@babel/helper-function-name": "7.0.0-beta.39",
  "@babel/helper-get-function-arity": "7.0.0-beta.39",
  "@babel/helper-hoist-variables": "7.0.0-beta.39",
  "@babel/helper-module-imports": "7.0.0-beta.39",
  "@babel/helper-module-transforms": "7.0.0-beta.39",
  "@babel/helper-optimise-call-expression": "7.0.0-beta.39",
  "@babel/helper-plugin-test-runner": "7.0.0-beta.39",
  "@babel/helper-regex": "7.0.0-beta.39",
  "@babel/helper-remap-async-to-generator": "7.0.0-beta.39",
  "@babel/helper-replace-supers": "7.0.0-beta.39",
  "@babel/helper-simple-access": "7.0.0-beta.39",
  "@babel/helper-transform-fixture-test-runner": "7.0.0-beta.39",
  "@babel/helper-wrap-function": "7.0.0-beta.39",
  "@babel/helpers": "7.0.0-beta.39",
  "@babel/node": "7.0.0-beta.39",
  "@babel/plugin-codemod-optional-catch-binding": "7.0.0-beta.39",
  "@babel/plugin-external-helpers": "7.0.0-beta.39",
  "@babel/plugin-proposal-async-generator-functions": "7.0.0-beta.39",
  "@babel/plugin-proposal-class-properties": "7.0.0-beta.39",
  "@babel/plugin-proposal-decorators": "7.0.0-beta.39",
  "@babel/plugin-proposal-do-expressions": "7.0.0-beta.39",
  "@babel/plugin-proposal-export-default-from": "7.0.0-beta.39",
  "@babel/plugin-proposal-export-namespace-from": "7.0.0-beta.39",
  "@babel/plugin-proposal-function-bind": "7.0.0-beta.39",
  "@babel/plugin-proposal-function-sent": "7.0.0-beta.39",
  "@babel/plugin-proposal-nullish-coalescing-operator": "7.0.0-beta.39",
  "@babel/plugin-proposal-numeric-separator": "7.0.0-beta.39",
  "@babel/plugin-proposal-object-rest-spread": "7.0.0-beta.39",
  "@babel/plugin-proposal-optional-catch-binding": "7.0.0-beta.39",
  "@babel/plugin-proposal-optional-chaining": "7.0.0-beta.39",
  "@babel/plugin-proposal-pipeline-operator": "7.0.0-beta.39",
  "@babel/plugin-proposal-throw-expressions": "7.0.0-beta.39",
  "@babel/plugin-proposal-unicode-property-regex": "7.0.0-beta.39",
  "@babel/plugin-syntax-async-generators": "7.0.0-beta.39",
  "@babel/plugin-syntax-class-properties": "7.0.0-beta.39",
  "@babel/plugin-syntax-decorators": "7.0.0-beta.39",
  "@babel/plugin-syntax-do-expressions": "7.0.0-beta.39",
  "@babel/plugin-syntax-dynamic-import": "7.0.0-beta.39",
  "@babel/plugin-syntax-export-default-from": "7.0.0-beta.39",
  "@babel/plugin-syntax-export-namespace-from": "7.0.0-beta.39",
  "@babel/plugin-syntax-flow": "7.0.0-beta.39",
  "@babel/plugin-syntax-function-bind": "7.0.0-beta.39",
  "@babel/plugin-syntax-function-sent": "7.0.0-beta.39",
  "@babel/plugin-syntax-import-meta": "7.0.0-beta.39",
  "@babel/plugin-syntax-jsx": "7.0.0-beta.39",
  "@babel/plugin-syntax-nullish-coalescing-operator": "7.0.0-beta.39",
  "@babel/plugin-syntax-numeric-separator": "7.0.0-beta.39",
  "@babel/plugin-syntax-object-rest-spread": "7.0.0-beta.39",
  "@babel/plugin-syntax-optional-catch-binding": "7.0.0-beta.39",
  "@babel/plugin-syntax-optional-chaining": "7.0.0-beta.39",
  "@babel/plugin-syntax-pipeline-operator": "7.0.0-beta.39",
  "@babel/plugin-syntax-throw-expressions": "7.0.0-beta.39",
  "@babel/plugin-syntax-typescript": "7.0.0-beta.39",
  "@babel/plugin-transform-arrow-functions": "7.0.0-beta.39",
  "@babel/plugin-transform-async-to-generator": "7.0.0-beta.39",
  "@babel/plugin-transform-block-scoped-functions": "7.0.0-beta.39",
  "@babel/plugin-transform-block-scoping": "7.0.0-beta.39",
  "@babel/plugin-transform-classes": "7.0.0-beta.39",
  "@babel/plugin-transform-computed-properties": "7.0.0-beta.39",
  "@babel/plugin-transform-destructuring": "7.0.0-beta.39",
  "@babel/plugin-transform-duplicate-keys": "7.0.0-beta.39",
  "@babel/plugin-transform-eval": "7.0.0-beta.39",
  "@babel/plugin-transform-exponentiation-operator": "7.0.0-beta.39",
  "@babel/plugin-transform-flow-comments": "7.0.0-beta.39",
  "@babel/plugin-transform-flow-strip-types": "7.0.0-beta.39",
  "@babel/plugin-transform-for-of": "7.0.0-beta.39",
  "@babel/plugin-transform-function-name": "7.0.0-beta.39",
  "@babel/plugin-transform-instanceof": "7.0.0-beta.39",
  "@babel/plugin-transform-jscript": "7.0.0-beta.39",
  "@babel/plugin-transform-literals": "7.0.0-beta.39",
  "@babel/plugin-transform-member-expression-literals": "7.0.0-beta.39",
  "@babel/plugin-transform-modules-amd": "7.0.0-beta.39",
  "@babel/plugin-transform-modules-commonjs": "7.0.0-beta.39",
  "@babel/plugin-transform-modules-systemjs": "7.0.0-beta.39",
  "@babel/plugin-transform-modules-umd": "7.0.0-beta.39",
  "@babel/plugin-transform-new-target": "7.0.0-beta.39",
  "@babel/plugin-transform-object-assign": "7.0.0-beta.39",
  "@babel/plugin-transform-object-set-prototype-of-to-assign": "7.0.0-beta.39",
  "@babel/plugin-transform-object-super": "7.0.0-beta.39",
  "@babel/plugin-transform-parameters": "7.0.0-beta.39",
  "@babel/plugin-transform-property-literals": "7.0.0-beta.39",
  "@babel/plugin-transform-property-mutators": "7.0.0-beta.39",
  "@babel/plugin-transform-proto-to-assign": "7.0.0-beta.39",
  "@babel/plugin-transform-react-constant-elements": "7.0.0-beta.39",
  "@babel/plugin-transform-react-display-name": "7.0.0-beta.39",
  "@babel/plugin-transform-react-inline-elements": "7.0.0-beta.39",
  "@babel/plugin-transform-react-jsx": "7.0.0-beta.39",
  "@babel/plugin-transform-react-jsx-compat": "7.0.0-beta.39",
  "@babel/plugin-transform-react-jsx-self": "7.0.0-beta.39",
  "@babel/plugin-transform-react-jsx-source": "7.0.0-beta.39",
  "@babel/plugin-transform-regenerator": "7.0.0-beta.39",
  "@babel/plugin-transform-reserved-words": "7.0.0-beta.39",
  "@babel/plugin-transform-runtime": "7.0.0-beta.39",
  "@babel/plugin-transform-shorthand-properties": "7.0.0-beta.39",
  "@babel/plugin-transform-spread": "7.0.0-beta.39",
  "@babel/plugin-transform-sticky-regex": "7.0.0-beta.39",
  "@babel/plugin-transform-strict-mode": "7.0.0-beta.39",
  "@babel/plugin-transform-template-literals": "7.0.0-beta.39",
  "@babel/plugin-transform-typeof-symbol": "7.0.0-beta.39",
  "@babel/plugin-transform-typescript": "7.0.0-beta.39",
  "@babel/plugin-transform-unicode-regex": "7.0.0-beta.39",
  "@babel/polyfill": "7.0.0-beta.39",
  "@babel/preset-env": "7.0.0-beta.39",
  "@babel/preset-env-standalone": "7.0.0-beta.39",
  "@babel/preset-flow": "7.0.0-beta.39",
  "@babel/preset-react": "7.0.0-beta.39",
  "@babel/preset-stage-0": "7.0.0-beta.39",
  "@babel/preset-stage-1": "7.0.0-beta.39",
  "@babel/preset-stage-2": "7.0.0-beta.39",
  "@babel/preset-stage-3": "7.0.0-beta.39",
  "@babel/preset-typescript": "7.0.0-beta.39",
  "@babel/register": "7.0.0-beta.39",
  "@babel/runtime": "7.0.0-beta.39",
  "@babel/standalone": "7.0.0-beta.39",
  "@babel/template": "7.0.0-beta.39",
  "@babel/traverse": "7.0.0-beta.39",
  "@babel/types": "7.0.0-beta.39",
  "babel-loader": "v8.0.0-beta.0",
}
`;
